### ISSUE TITLE
Enable React.js

### ISSUE THREAD

**Issue by lubril** [lubril/perfin#1]
Could use React.js for its cool [react-grid-layout](https://strml.github.io/react-grid-layout/examples/0-showcase.html) which could be the base for a chart dashboard just like this: [https://react-dashboard-demo.cube.dev/#/](https://react-dashboard-demo.cube.dev/#/)
How to enable React.js for Symfony: [https://medium.com/@lmatte7/how-to-use-react-with-symfony-4-88fb27abf5e5](https://medium.com/@lmatte7/how-to-use-react-with-symfony-4-88fb27abf5e5)

**Comment by geropl** [comment]
Cool, `react-grid-layout` looks promising! If we use it separately we would have to store the charts/dashboard layout info ourselves, but that should not be to hard. :slightly_smiling_face: 

I had a look at the links you provided, and both suggest using the `.twig` templates to inject (static) values into the JS part, which seems a bit arcane and inconvenient too me.

One could also:
 - describe all frontend in react (Typescript)
 - use a static site generator like [gatsbyjs](https://www.gatsbyjs.org/) to generate static files (HTML, CSS, JS) from it
 - serve those static files with laravel (or any webserver for that matter)
 - write a simple API using laravel which we call from frontend (e.g., REST for starters and simplicity)

Maybe I find some time to prototype what I mean. :roll_eyes:

**Comment by geropl** [comment]
I managed to hack together #2 as a first step.

**Comment by geropl** [comment]
Hum. The more I think about it... what do we need a backend at all? :smile: And package with electron?

**Comment by lubril** [comment]
really, I'm open to anything... and a little scared that all that means we'll scrap symfony altogether...

So just let me get this straight, you suggest

- we'll provide the existing REST API with symfony
- have laravel serving our static pages which call the REST API (doesn't that mean we'd always have 2 running servers)
- use gatsbyjs to retrieve data and generate those static webpages

and alternatively we could use electron which... does what?! I skimmed over some explanations but it's still pretty unclear to me what it actually does.
Basically every part of this tool stack is new to me 😅
